internal/controller/controlplane/k0s_controlplane_controller.go

```diff
@@ -587,10 +594,7 @@ func (c *K0sController) deleteK0sNodeResources(ctx context.Context, cluster *clu
 	return nil
 }
 
-func (c *K0sController) createBootstrapConfig(ctx context.Context, name string, _ *clusterv1.Cluster, kcp *cpv1beta1.K0sControlPlane, machine *clusterv1.Machine, clusterName string) error {
-
-	k0sConfigSpec := kcp.Spec.K0sConfigSpec.DeepCopy()
-	k0sConfigSpec.Args = uniqueArgs(k0sConfigSpec.Args)
+func (c *K0sController) createBootstrapConfig(ctx context.Context, name string, k0sConfigSpec *bootstrapv1.K0sConfigSpec, kcp *cpv1beta1.K0sControlPlane, machine *clusterv1.Machine, clusterName string) error {
```

Comment:
Sorry, I'm not fully following. We update the annotation on each change and then use the value from annotation to update the bootstrap config, right?